src/core/lib/transport/metadata.h

```diff
@@ -109,7 +109,11 @@ struct grpc_mdelem {
                               (uintptr_t)GRPC_MDELEM_STORAGE_INTERNED_BIT))
 
 /* Unrefs the slices. */
-grpc_mdelem grpc_mdelem_from_slices(grpc_slice key, grpc_slice value);
+grpc_mdelem grpc_mdelem_from_slices(const grpc_slice& key,
```

Comment:
Is it actually helpful to pass the slices in as const references?  My understanding is that the `grpc_slice` API was designed such that you can pass slices around by value, since they're essentially no bigger than a pointer.  But I haven't actually looked at the implementation to verify that.

Reply:
I'm happy to remove this, but a slice is essentially 4 pointers (one for refcount, and 4 for the inlined space) unlike a std::string_view which is 2 pointers. Passing two slices would 8 stores. IMHO, there will be no harm in passing as const ref.